lib/slack/post.rb

```diff
@@ -33,7 +33,6 @@ def self.post_with_attachments(message, attachments, chan = nil, opts = {})
 
 			http = Net::HTTP.new(uri.host, uri.port, config[:proxy_host], config[:proxy_port])
 			http.use_ssl = true
-			http.ssl_version = :TLSv1_2
```

Comment:
This is more a flaw with the SSL/TLS protocols than a fault of the code here but [as the docs hint](https://ruby-doc.org/stdlib-2.7.1/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html#method-i-ssl_version-3D) we should be using `min_version = :TLSv1_2` to prevent downgrade attacks

Reply:
Nice catch! Updated

```
[1] pry(Slack::Post)> http.set_debug_output($stdout)
=> #<IO:<STDOUT>>
[2] pry(Slack::Post)> http.min_version = OpenSSL::SSL::TLS1_2_VERSION
=> 771
[3] pry(Slack::Post)> continue
opening connection to REDACTED.slack.com:443...
opened
starting SSL for REDACTED.slack.com:443...
SSL established, protocol: TLSv1.3, cipher: TLS_AES_128_GCM_SHA256
<- "POST /services/hooks/incoming-webhook?token=REDACTED HTTP/1.1\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nContent-Type: application/json\r\nConnection: close\r\nHost: REDACTED.slack.com\r\nContent-Length: 62\r\n\r\n"
<- "{\"channel\":\"#testing\",\"text\":\"no tlsv_\",\"username\":\"im.kevin\"}"
-> "HTTP/1.1 200 OK\r\n"
-> "date: Mon, 04 May 2020 22:49:50 GMT\r\n"
-> "server: Apache\r\n"
-> "strict-transport-security: max-age=31536000; includeSubDomains; preload\r\n"
-> "referrer-policy: no-referrer\r\n"
-> "vary: Accept-Encoding\r\n"
-> "x-slack-backend: h\r\n"
-> "x-frame-options: SAMEORIGIN\r\n"
-> "content-encoding: gzip\r\n"
-> "access-control-allow-origin: *\r\n"
-> "content-length: 22\r\n"
-> "connection: close\r\n"
-> "content-type: text/html\r\n"
-> "x-via: haproxy-www-kjn3\r\n"
-> "\r\n"
reading 22 bytes...
-> "\x1F\x8B\b\x00\x00\x00\x00\x00\x00\x03\xCB\xCF\x06\x00G\xDD\xDCy\x02\x00\x00\x00"
read 22 bytes
Conn close
```